modules/ce_google_docs/src/Controller/GoogleDocsController.php

```diff
@@ -0,0 +1,190 @@
+        $response = $client->request('GET', $src);
+        $name = str_replace(" ", "_", substr($response->getHeader('Content-Disposition')[0], 17, -1));
+        $file = system_retrieve_file($src, 'public://' . $name, TRUE, $replace = FILE_EXISTS_RENAME);
+        $new_src_url = explode($_SERVER['HTTP_HOST'], $file->url())[1];
+        $body = str_replace($src, $new_src_url, $body);
+      }
+    }
+
+    $body_feild = '';
+    $fields = \Drupal::entityManager()->getFieldDefinitions('node', $request->request->get('type'));
+    foreach ($fields as $field => $value) {
+      if ($value->getType() == "text_with_summary") {
+        $body_feild = $field;
+        break;
+      }
+    }
+
+    $node = Node::create([
+      'type' => $request->request->get('type'),
+      'created' => \Drupal::time()->getRequestTime(),
+      'uid' => $request->request->get('user'),
+      'title' => $request->request->get('title'),
+      $body_feild => [
+        'summary' => $request->request->get('summary'),
+        'value' => $body,
+        'format' => 'full_html',
+      ],
+      'status' => $request->request->get('status'),
+    ]);
+    // Saving the node.
+    $node->save();
+
+    return new JsonResponse(['nid' => $node->id()]);
+  }
+
+  /**
+   * To list all the eligible Drupal Content Type.
+   *
+   * @param Symfony\Component\HttpFoundation\Request $request
+   *   HttpRequest with apiKey parameter as API Key.
+   *
+   * @return Symfony\Component\HttpFoundation\JsonResponse
+   *   JsonResponse with Content Type(s).
+   */
+  public function listContentType(Request $request) {
+    $config = \Drupal::config('ce_google_docs.settings');
+    if ($request->query->get('apiKey') != $config->get('api_key')) {
+      return new JsonResponse(['error' => 'Invalid API Key.']);
+    }
+    $content_type = [];
+    $nodes = entity_get_bundles('node');
+    // Loop through each content type.
+    foreach ($nodes as $type => $node) {
+      // Get all fields on the content type.
+      $fields = \Drupal::entityManager()->getFieldDefinitions('node', $type);
+      foreach ($fields as $field => $value) {
+        if ($value->getType() == "text_with_summary") {
+          array_push($content_type, [
+            'id' => $type,
+            'name' => $node['label'],
+          ]);
+          break;
+        }
+      }
+    }
+
+    return new JsonResponse($content_type);
+  }
+
+  /**
+   * To list all the Drupal Users.
+   *
+   * @param Symfony\Component\HttpFoundation\Request $request
+   *   HttpRequest with apiKey parameter as API Key.
+   *
+   * @return Symfony\Component\HttpFoundation\JsonResponse
+   *   JsonResponse with User's ID and Name.
+   */
+  public function listUser(Request $request) {
+    $config = \Drupal::config('ce_google_docs.settings');
+    if ($request->query->get('apiKey') != $config->get('api_key')) {
+      return new JsonResponse(['error' => 'Invalid API Key.']);
+    }
+    $users = user_load_multiple();
```

Comment:
There should be some way of user authentication else anyone can publish the content from any user. Only admin can have access to publish content from any user but it shouldn't be practised.